go/parquet/pqarrow/path_builder.go

```diff
@@ -555,7 +555,6 @@ type multipathLevelResult struct {
 }
 
 func (m *multipathLevelResult) Release() {
-	m.leafArr.Release()
```

Comment:
`pathBuilder.addTerminalInfo` itself does call `paths.primitiveArr.Retain` (and a corresponding `Release`), so I think this is safe to remove? This would also explain why the leak only happens with nested types (because they have an extra array not released by this `leafArr.Release`)